main.js

```diff
@@ -1337,10 +1337,13 @@ function sendSSH(method, cb) {
             pass: adapter.config.password
         });
 
-        ssh.exec(`echo "${adapter.config.password}"|sudo -S ${method} now`,{
+        //old code ssh.exec(`echo \'${adapter.config.password}\n\'|sudo -S ${method}`,{
+        const sshcmd= 'echo \''+adapter.config.password+'\' | sudo -S ' + method;
```

Comment:
Das fixt aber keine Probleme mit sonderzeichen! ... bzw effektiv ists nicht wirklich ein change zu vorher - nur ein zeilenumbruch der eh zuviel war ist nicht da? Sonderzeichen wie ein ' machen immer noch issues

Reply:
As discussed at telegram:
The main change is that the shell command now uses single qutes to echo the password which protects most special characters (i.e. a dollar sign).

Todo:
Find a solution for very special characters, i.e. the single quote and mayby the \ escape character

Reply:
Kurzes google sagt das escapen von ' mit einem \ gehen sollte. also quasi oben ein 

```
adapter.config.password.replace(/'/g, '\\\'');
```

könnte vllt schon Lösung sein

Reply:
Da scheint google nur tw. recht zu haben.
In der klassischen Shell bzw Bash wird innerhalb von eines ' Strings nichts ersetzt.
Allerdings gibt es eine bash Erweiterung die das Verhalten bei $'xxxx' ändert. Hier wird dann einiges, u.a. auch \' ersetzt. Wichtig ist da das $ am Anfang.

Egal - Test auf Pi4 hat ergeben:
```
pi@pi4:~ $
pi@pi4:~ $ echo 'x\\\'x'
> ^C
pi@pi4:~ $ echo $'x\'x'
x'x
pi@pi4:~ $ echo 'x\'x'
> ^C
pi@pi4:~ $
```
soll heißen, es funktioniert nur die $ Variante. Die Replaced dann aber gelich wieder mehr ...

Reply:
Was geht ist mix aus einfachem und doppeltem Hochkomma in Verbindung imit impliziter Stringcancatenation:
```
pi@pi4:~ $ echo 'x'"'"'x'
x'x
pi@pi4:~ $
```

also das ' durch '"'"' ersetzten - und sich dabei nicht verzählen :-)